include/Mathematical/SquareMatrix.hpp

```diff
@@ -311,9 +327,81 @@ class SquareMatrix : public MatrixX<_Scalar> {
 
         return value;
     }
-};
 
 
+    /**
+     *  @return an non-pivoted LU decomposition in an array, with L at position 0 and U on position 1 of the array.
+     *   warning: Pivoting is required to ensure that the decomposition is stable. Eigen3 provides partial and full pivot modules.
```

Comment:
```suggestion
     *   Warning: pivoting is required to ensure that the decomposition is stable. Eigen3 provides partial and full pivot modules
```

So could you add to the documentation when I should use this decomposition and when I should use Eigen?